server/viame_server/__init__.py

```diff
@@ -32,11 +32,21 @@ def load_pipelines():
 
     allowed = r"^detector_.+|tracker_.+"
     disallowed = r".*local.*|detector_svm_models.pipe|tracker_svm_models.pipe"
-    return [
+    pipelist = [
         path.name
         for path in pipeline_path.glob("./*.pipe")
         if re.match(allowed, path.name) and not re.match(disallowed, path.name)
     ]
+    pipedict = {
+        'detector': {'pipes': [], 'description': '',},
+        'tracker': {'pipes': [], 'description': '',},
+    }
+    for pipe in pipelist:
+        nameparts = pipe.replace('.pipe', '').split('_')
+        name = ' '.join(nameparts[1:])
+        type = nameparts[0]
+        pipedict[type]['pipes'].append({'name': name, 'type': type, 'pipe': pipe})
+    return pipedict
```

Comment:
Process and parse the pipelines server-side.  Protect the client code from having to know the particulars of pipe naming conventions.  

For example, if convention switched from `TYPPE_PIPENAME.pipe` for determining pipe type to subfolders and folder name, the client code shouldn't be aware of this change.